File: C++/CivitasAcademic.cpp
// Import library yang dibutuhkan
#include "Human.cpp" // Import file Human.cpp

using namespace std; // Menggunakan standard namespace

class CivitasAcademic : public Human // Membuat class CivitasAcademic (child) yang mengextends ke class Human (parent)
{
protected:             // Membuat atribut berjenis protected
    string university; // Untuk menyimpan nama universitas mahasiswa
    string email;      // Untuk menyimpan email mahasiswa

public:
    /* Konstruktor */
    CivitasAcademic() : Human() // Membuat konstruktor tanpa isian, class yang di-extends akan diinstansiasi terlebih dahulu
    {
        this->university = "";
        this->email = "";
    }

    CivitasAcademic(long long identity, string name, char gender, string university, string email) : Human(identity, name, gender) // Membuat konstruktor dengan isian dari parameter
    {
        this->university = university;
        this->email = email;
    }

    /* Setter dan Getter untuk setiap atribut di dalam class CivitasAcademic */
    void setUniversity(string university)
    {
        this->university = university;
    }

    void setEmail(string email)
    {
        this->email = email;
    }

    string getUniversity()
    {
        return this->university;
    }

    string getEmail()
    {
        return this->email;
    }

    /* Destruktor */
    ~CivitasAcademic()
    {
    }
};

File: Python/civitasacademic.py
from human import Human # Import class Human dari file human.py

class CivitasAcademic(Human): # Deklarasi class CivitasAcademic (child class dari Human)
    _university = "" # Untuk menyimpan nama universitas, atribut berjenis protected
    _email = "" # Untuk menyimpan email mahasiswa, atribut berjenis protected

    def __init__(self, identity = 0, name = "", gender = '', university = "", email = ""): # Membuat konstruktor
        super().__init__(identity, name, gender) # Menginisialisasi setiap atribut yang ada di dalam konstruktor parent class Human
        self._university = university # Menginisialisasi atribut _university
        self._email = email # Menginisialisasi atribut _email

    def setUniversity(self, university): # Setter untuk atribut _university
        self._university = university # Menginisialisasi atribut _university

    def setEmail(self, email): # Setter untuk atribut _email
        self._email = email # Menginisialisasi atribut _email

    def getUniversity(self): # Getter untuk atribut _university
        return self._university # Mengembalikan nilai atribut _university

    def getEmail(self): # Getter untuk atribut _email
        return self._email # Mengembalikan nilai atribut _email
